### DISCUSSION TITLE
再次对issue反馈进行建议（原discussion: #1592）

### DISCUSSION THREAD

**Discussion by iPodtouch6557** [Meloong-Git/PCL#1856] [讨论]
**近日来开始有人乱交 issue**，例如：明知自己已经交过一个了，还要交一个重复的；或者是把 bug 交成优化；甚至有人为了活跃橙瞎交 issue……

### 在这里，我对这些相关问题进行建议

> 可以参考 #1592 和 https://github.com/tvvocold/How-To-Ask-The-Smart-Way

2.请选择正确的 category，是 bug 就是 bug ，是优化就是优化，不要顶着个 bug 交优化
3.好的标题也很重要
>**1.请为你的 issue 撰写一个具体的标题，不要像“一个崩溃”“出错了”这样不具体的标题，如果你的内容长到会致使标题过长，请进行适当取舍，把一些内容放到`描述`一栏**

>**2.请把你的标题写的合适一点，属于 bug 的话就不要说优化一类的，更不要标题与内容不符**

4.如果你反馈了一个又类似bug又类似优化的内容，而你又交了一个优化并被当成优化采纳，请不要为你拿不到活跃橙打抱不平，并且在自己已完成或者别人已完成的内容底下讨论 off topic，这样会使 issue 被锁 conversation。事实上只要你懂一点相关代码，其实你也可以交一个 pr，我就是靠一个 pr 拿到活跃橙的……

**注： @Chiloven945 对此 discussion 内容提出建议**，具体可看 https://github.com/Hex-Dragon/PCL2/discussions/1856#discussioncomment-5906546

**Comment by Chiloven945** [comment]
可以加点东西
标题得是有概括性的，其中**特别**注意尽量不要有符号，像是问号、逗号、句号等等，有些人写问号可能是因为拿不定这个issue是否妥当，这种情况可以先交discussion问问大家之后再转到issue，当然也有人可能是因为习惯
一个issue也只能描述一个问题，不要出现多个问题在一个issue的情况

**Reply by Chiloven945** [reply]
还有这discussion也有点乱，有些人就是在提问但是交的却是讨论

**Reply by iPodtouch6557** [reply]
已改，但是懒得亲自敲进去，就加了个链接（

**Reply by Chormore-EDGE** [reply]
> 还有这讨论也有点乱，有些人就是在提问但是交的却是讨论

 有点时候甚至不知道这是讨论还是issue，但也没办法，如果交了一个和主issue相关的issue可能会被说重复...~例如版本号的那个.....~
交：issue增多，大部分可能为重复
不交：讨论如issue~心理也痒痒~(

**Reply by WForst-Breeze** [reply]
> 还有这discussion也有点乱，有些人就是在提问但是交的却是讨论

这我个人感觉是小问题，毕竟这两者在实际情况下有时也会分分不太清（可以说，提问也是引发讨论的一种🤔），有些问题也可以在讨论的过程中得到答案。能帮助人们解决问题，目的就达到了🥳

**Reply by Chormore-EDGE** [reply]
实际上这种现象没法改变，有些人(比如我)容易在issue中的讨论想出另一个issue但有可能想出重复内容
之后又回到了我的交或者不交的问题上了(

**Reply by Chiloven945** [reply]
> > 还有这discussion也有点乱，有些人就是在提问但是交的却是讨论
> 
> 这我个人感觉是小问题，毕竟这两者在实际情况下有时也会分分不太清（可以说，提问也是引发讨论的一种🤔），有些问题也可以在讨论的过程中得到答案。能帮助人们解决问题，目的就达到了🥳

这里的意思我到感觉是讨论是抛出一个问题再具体地对这个问题发表自己的看法，更倾向于没有正确答案的问题~比如说吃咸豆腐脑还是甜豆腐脑~，而提问则更加聚焦于问答上，比如说这个怎么做什么的

**Reply by iPodtouch6557** [reply]
> 这里的意思我到感觉是讨论是抛出一个问题再具体地对这个问题发表自己的看法，更倾向于没有正确答案的问题比如说吃咸豆腐脑还是甜豆腐脑，而提问则更加聚焦于问答上，比如说这个怎么做什么的

我个人更认为不适合交issue的建议，就比如金数据和github issue，或者伯特的相关建议适合交discussions；or一些初步的疑问也可以先交discussions，但得讲明这是初步的一个提交，要不然就是被误认为那种乱交
~而且萌新们一般是初来github不会选category，瞎点点出来的，我最早就是~

**Comment by Chormore-EDGE** [comment]
~虽然我也是抱着集卡的心态去提交issue的~不过乱交issue真的很过分
建议置顶

**Comment by ghost** [comment]
事实上，我觉得这个讨论就是针对@deep-dark-forest提出的，他不就是交了个又像优化又像bug的东西，但是，对于活跃橙这件事其实你可以自己想想自己的经历，在之前的问题中你也是因为交了优化而没有得到活跃橙，现在，你看到了别人交了个擦边球issue得了活跃橙，自己心里可能有些不舒服，这是可以理解的……但是，无论如何都不能以这种方式报复别人吧

**Reply by iPodtouch6557** [reply]
没有针对，是广泛指那些刚来github啥也不懂想给pcl反馈点问题的人，我认为这已经算一个群体了
**而且我在这里非常严厉的告诉你，我根本就没有任何要报复的意思**，我早在这个人拿到活跃橙之前，就已经靠一个pr拿到了，此前我一直没想到我能靠一个pr拿到这玩意，所以其实他们拿不拿的到还是咋地我其实不太在意……

**Reply by ghost** [reply]
> 没有针对，是广泛指那些刚来github啥也不懂想给pcl反馈点问题的人，我认为这已经算一个群体了
> **而且我在这里非常严厉的告诉你，我根本就没有任何要报复的意思**，我早在这个人拿到活跃橙之前，就已经靠一个pr拿到了，此前我一直没想到我能靠一个pr拿到这玩意，所以其实他们拿不拿的到还是咋地我其实不太在意……
> 

即便这样，你也是在@deep-dark-forest发起了那个issue后你才发起的这个讨论，也许我报复这个词用的有点过，但是至少也有针对的意思吧，毕竟这里面所有内容都感觉像是在说他

**Reply by iPodtouch6557** [reply]
> 即便这样，你也是在@deep-dark-forest发起了那个issue后你才发起的这个讨论，也许我报复这个词用的有点过，但是至少也有针对的意思吧，毕竟这里面所有内容都感觉像是在说他

确实有由着这个为范本说事情的意思，但真的不是针对哈
请你不要再随意去曲解我的意思了……

**Comment by LTCatt** [comment]
关于 Bug/优化 Category 的问题，如果我觉得选错了，我会在标记为 `处理中` 的时候就一并把 Tag 纠正。
有一部分优化确实也可以视作 Bug，这主要是看 **“这个玩意儿有没有按照我预期的方式起到预期的功能”**。例如 #1806 中，它的预期功能是能让用户去购买 MC，但实际上这个链接不能，那么 **没有起到预期功能，它就可以视作 Bug。** 因为提交者自己选了优化标签，它某种意义上也可以说是优化，所以我没有改 Tag 罢了 =。=
对有的功能，**我的预期** 可能和大家的理解不太一样，所以有的优化在我眼里确实是 Bug，而有的 “Bug” 我会觉得只是没有处理得更好，但还是达到了预期的，所以会判成优化。如果在这方面有异议可以直接在原 issue 下回复。

---

关于活跃橙，**为了活跃橙交 issue 是可以接受的**，这条规则本身就是为了鼓励社区积极汇报 Bug，特别是一些罕见 Bug。
确实偶尔会出现故意交一些莫名其妙的 “bug” 的情况，我也在相关 issue 下面说过了，自己刻意触发导致的问题（例如手动将关键文件设置为只读）不会被受理。这类 issue 才是真正不想要的，不过目前也没有产生什么影响。

---

总而言之，GitHub 是为了**解决问题**的，既然问题解决了，也不用太在意过程嘛。感谢大家啦！

**Reply by iPodtouch6557** [reply]
好的，我尝试搞掉
